src/NzbDrone.Core/Download/Clients/Porla/Porla.cs

```
@@ -0,0 +1,281 @@
+
+            // var sessionSettings = _proxy.GetSessionSettings(Settings);
+
+            var status = new DownloadClientInfo
+            {
+                IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                RemovesCompletedDownloads = false, // TODO: I don't think porla has config for this, it feels like it should
+            };
+
+            var savePath = ((presetEffectiveSettings?.SavePath.IsNullOrWhiteSpace() ?? true) ? Settings.TvDirectory : presetEffectiveSettings?.SavePath) ?? "";
+            var destDir = new OsPath(savePath);
+            if (!destDir.IsEmpty)
+            {
+               status.OutputRootFolders = new List<OsPath> { _remotePathMappingService.RemapRemoteToLocal(Settings.Host, destDir) };
+            }
+
+            return status;
+        }
+
+        /// <summary> Converts a RemoteEpisode into a list of <em>starr</em> tags </summary>
+        /// <see cref="RemoteEpisode"/>
+        private static IList<string> ConvertRemoteEpisodeToTags(RemoteEpisode remoteEpisode)
+        {
+            var tags = new List<string>
+            {
+                $"starr.series={remoteEpisode.Series.CleanTitle}",
+                $"starr.season={remoteEpisode.MappedSeasonNumber}",
+                $"starr.tvdbid={remoteEpisode.Series.TvdbId}",
+                $"starr.imdbid={remoteEpisode.Series.ImdbId}",
+                $"starr.tvmazeid={remoteEpisode.Series.TvMazeId}",
+                $"starr.year={remoteEpisode.Series.Year}"
+            };
+            return tags;
+        }
+
+        // NOTE: If the torrent already exists in the client, it'll fail with error code -3
+        // {
+        //   "code": -3,
+        //   "data": null,
+        //   "message": "Torrent already in session 'default'"
+        // }
+        // IDK If I should deal with that...
```

If it already exists then we either need to throw an error (why isn't Sonarr tracking it?) or treat it as successful, pros and cons to both.